core/elbv2.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	"github.com/aws/aws-sdk-go/service/elbv2"
+)
+
+// Returns a list of ELB (v2) - also known as ALB - based on the input from the different regions
+func (c *Connector) GetLoadBalancersV2(input *elbv2.DescribeLoadBalancersInput) ([]*elbv2.DescribeLoadBalancersOutput, error) {
```

Comment:
Ups, I didn't want to mean that.

For such reason, I was complaining about AWS SDK returns a generic error than their specific error interfaces on methods that they can only return one specific error type. If their function can return multiple interface type of errors in a function, you have to choose the narrowest interface which is satisfied by all of those, obviously the `error` interface should be satisfied by all of them, so there is always that chance.

I would like to remember the right terminology but if I don't remember bad (I'm shit remember names and terminology), a function which is more extensible must be covariance in their arguments and contravariance in their return values ([Covariance and contravariance (computer science)](https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science))

What's standard is to return an value which satisfies the `error` interface. What I wanted to mean is that returning `type struct` rather than a `type interface` has more limitations for the callers and from the creator point of view, could lead to export some `struct fields` that they can be removed later on, but the callers are relying on them, so it's more prone to break compatibility than using interfaces.

For this reason, I suggest to return the interface.